### PR TITLE
DSD-520: Adding the title element in the SVG icons

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-520](https://jira.nypl.org/browse/DSD-520)

## This PR does the following:
- Adds a `title` element to the dynamically generated SVGs in the `Icon` component. This is done by the svgr package but needs to be configured in the Storybook webpack config and the DS rollup config.
- These `title` HTML elements are not generated in unit tests so it's hard to test.
- A `title` element is _always_ generated and the default text is the name of the icon along with "icon". For example, `download icon` or `error_filled icon`. I think it's good enough to use the icon code name and suggest that devs pass in readable text for the `title` prop.

## How has this been tested?

Storybook.

### Acessibility Note

`title` attributes are normally used in HTML elements but this does not work for the `svg` element. In order to make it accessible, a `title` _HTML element_ should be rendered inside the `svg` element.

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] View [the example in Storybook]()

### CODE REVIEW

File: src/components/Icons/Icon.stories.mdx [hunk 1]
```diff
@@ -186,6 +186,13 @@ export const allIconsType = (list) => (
   <ul style={{ listStyle: "none" }}>{list}</ul>
 );
 
+## Accessibility
```
Comment (bigfishdesign13) [inline src/components/Icons/Icon.stories.mdx:189]: Can you also add something to the `title` prop's description stressing the importance of the prop when it comes to accessibilty?

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
The `title` prop could use some more info detailing how the prop impacts accessibility.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr744-lopzoddhyyp7dikz5s2mpyej2jsbp2j7.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/617b1f519468f8221b21d506

**Review by bigfishdesign13 — APPROVED** [review]
Awesome.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr744-lopzoddhyyp7dikz5s2mpyej2jsbp2j7.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/617b1f519468f8221b21d506

**Pull request merged**